crates/compiler/can/src/operator.rs

```diff
@@ -544,6 +548,87 @@ fn desugar_field<'a>(
     }
 }
 
+fn desugar_loc_patterns<'a>(
+    arena: &'a Bump,
+    loc_patterns: &'a [Loc<Pattern<'a>>],
+) -> &'a [Loc<Pattern<'a>>] {
+    Vec::from_iter_in(
+        loc_patterns.iter().map(|loc_pattern| Loc {
+            region: loc_pattern.region,
+            value: desugar_pattern(arena, loc_pattern.value),
+        }),
+        arena,
+    )
+    .into_bump_slice()
+}
+
+fn desugar_loc_pattern<'a>(
+    arena: &'a Bump,
+    loc_pattern: &'a Loc<Pattern<'a>>,
+) -> &'a Loc<Pattern<'a>> {
+    arena.alloc(Loc {
+        region: loc_pattern.region,
+        value: desugar_pattern(arena, loc_pattern.value),
+    })
+}
+
+// Find `OptionalField`s and desugar the default value exprs. Also discard `SpaceBefore` and
+// `SpaceAfter` nodes so that patterns can be destructured over multiple lines.
```

Comment:
Slight preference for removing this comment. I think it's pretty clear what the function does from the body, and it's pretty likely it will grow out of date if the logic is ever updated.

Reply:
Makes sense, removed